samuroi/util/mask_generator/create_masks.py

```diff
@@ -0,0 +1,196 @@
+import numpy as np
+from scipy import ndimage
+from skimage import segmentation
+
+
+class MaskGenerator(object):
+    """
+    Parent class for generating masklists and manually correcting them
+    """
+    def __init__(self, blob_image, raw_image, com=None):
+
+        self.raw_image = raw_image
+        self.blob_image = blob_image
+
+        if com is None:
+            self.com = get_blob_centers(self.blob_image)
+        else:
+            self.com = com
+
+        self.putative_nuclei_image, _ = remove_small_blobs(self.blob_image, self.com)  # remove non-overlapping centers
+
+    def append_com(self, c):
+        x = self.com.tolist()
+        x.append(c)
+        self.com = np.array(x)
+
+    def remove_com(self, c):
+        x = self.com.tolist()
+        x.remove(c)
+        self.com = np.array(x)
+
+    def update(self):
+        pass
+
+    def create_roi_masks(self, seg_type):
+        pass
+
+
+class DonutCells(MaskGenerator):
+    """
+    Child class to handle GCaMP data - i.e. donut shaped rois, with a dark nucleus
+    """
+    def __init__(self, raw_image, putative_nuclei_image, putative_somata_image, com=None):
+        super(DonutCells, self).__init__(putative_nuclei_image, raw_image, com)
+        self.putative_somata_image = putative_somata_image
+        self.putative_nuclei_image = ndimage.binary_fill_holes(self.putative_nuclei_image)
+        self.watershed_image = np.logical_or(self.putative_nuclei_image, self.putative_somata_image)
+        self.segmentation_labels = calculate_distance(self.watershed_image, self.com)
+        self.labelled_nuclei = calculate_distance(self.putative_nuclei_image, self.com)
+        self.roi_masks = create_roi_masks(self.com, self.putative_nuclei_image, self.putative_somata_image)
+
+    def update(self):
+        self.putative_nuclei_image, _ = remove_small_blobs(self.putative_nuclei_image, self.com)
+        self.segmentation_labels = calculate_distance(self.watershed_image, self.com)
+        self.roi_masks = create_roi_masks(self.com, self.putative_nuclei_image, self.putative_somata_image)
+
+
+class BlobCells(MaskGenerator):
+    """
+    Child class for blob cells - i.e. bolus loaded cells. Nucleus also fluorescent
+    """
+
+    def __init__(self, segmentation_layer, raw_image):
+        super(BlobCells, self).__init__(segmentation_layer, raw_image)
+        self.watershed_image = self.putative_nuclei_image
+        self.segmentation_labels = calculate_distance(self.watershed_image, self.com)
+        self.roi_mask_list = self.create_roi_masks()
+
+    def create_roi_masks(self):
+        roi_mask_list = []
+        for i in range(np.max(self.segmentation_labels)):
+            idx = np.where(self.segmentation_labels*self.putative_nuclei_image == i+1)
+            x = np.vstack((idx[0], idx[1]))
+            m = np.reshape(x.T, (len(idx[0]), 2))
+            roi_mask_list.append(m)
+
+        return roi_mask_list
+
+
+def get_blob_centers(segmentation_layer, iterations=3):
```

Comment:
please rename, i think this essentially calculates the center of masss. this should be reflected in the function name.